bpf/init.sh

```diff
@@ -432,6 +434,9 @@ if [ "${TUNNEL_MODE}" != "<nil>" ]; then
 		ip link add name $ENCAP_DEV address $(rnd_mac_addr) type $TUNNEL_MODE external || encap_fail
 	}
 	ip link set $ENCAP_DEV mtu $MTU || encap_fail
+	if [ "${TUNNEL_PORT}" != "<nil>" ]; then
+		ip link set $ENCAP_DEV dstport $TUNNEL_PORT || encap_fail
```

Comment:
It looks this won't work, despite that being non-evident from the source code (https://github.com/cilium/cilium/issues/15956#issuecomment-878342725):
```
2021-07-15T08:46:49.775523869Z level=warning msg="+ ENCAP_DEV=cilium_vxlan" subsys=datapath-loader
2021-07-15T08:46:49.775526869Z level=warning msg="+ ip link show cilium_vxlan" subsys=datapath-loader
2021-07-15T08:46:49.775530169Z level=warning msg="4: cilium_vxlan: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 1000" subsys=datapath-loader
2021-07-15T08:46:49.775533770Z level=warning msg="    link/ether da:9e:0d:f9:97:0f brd ff:ff:ff:ff:ff:ff" subsys=datapath-loader
2021-07-15T08:46:49.775536870Z level=warning msg="+ ip link set cilium_vxlan mtu 1500" subsys=datapath-loader
2021-07-15T08:46:49.775540170Z level=warning msg="+ '[' 8472 '!=' '<nil>' ']'" subsys=datapath-loader
2021-07-15T08:46:49.775543370Z level=warning msg="+ ip link set cilium_vxlan dstport 8472" subsys=datapath-loader
2021-07-15T08:46:49.775547171Z level=warning msg="Error: either \"dev\" is duplicate, or \"dstport\" is a garbage." subsys=datapath-loader
```

Reply:
Looks like I just need to add `type vxlan` here also.

Reply:
The parse accepts the option, but it errors out:
```
2021-07-15T09:20:49.108884244Z level=warning msg="+ ip link set cilium_vxlan type vxlan dstport 8472" subsys=datapath-loader
2021-07-15T09:20:49.108888544Z level=warning msg="Error: Cannot change port." subsys=datapath-loader
```

Reply:
Will try to recreate the interface on port change, in any case port change would require tunnel reestablishment, so for the user who choses to change the port a disruption may seem reasonable. I don't know if we might need to add a flag, or if recreation of the interface might have other side-effects, let's discuss that here :)

Reply:
@borkmann is there anything that needs to change in the datapath in the even of interface re-creation? Just wondering in case this might had been unforeseen previously.

Reply:
I've submitted [a patch to fix iproute2](https://patchwork.kernel.org/project/netdevbpf/patch/20210716153557.10192-1-errordeveloper@gmail.com/).